mmseg/models/backbones/twins.py

```diff
@@ -0,0 +1,646 @@
+                 sr_ratios=[8, 4, 2, 1],
+                 extra_norm=False,
+                 init_cfg=None):
+        super(PCPVT, self).__init__(init_cfg=init_cfg)
+        self.num_classes = num_classes
+        self.depths = depths
+
+        # patch_embed
+        self.patch_embeds = ModuleList()
+        self.pos_drops = ModuleList()
+        self.blocks = ModuleList()
+
+        for i in range(len(depths)):
+            if i == 0:
+                input_size = img_size
+                self.patch_embeds.append(
+                    PatchEmbed(
+                        in_channels=in_channels,
+                        embed_dims=embed_dims[i],
+                        conv_type='Conv2d',
+                        kernel_size=patch_size,
+                        stride=patch_size,
+                        padding='corner',
+                        norm_cfg=norm_cfg,
+                        input_size=input_size,
+                        init_cfg=None))
+            else:
+                patch_size = 2
+                input_size = img_size // patch_size // 2**(i - 1)
+                self.patch_embeds.append(
+                    PatchEmbed(
+                        in_channels=embed_dims[i - 1],
+                        embed_dims=embed_dims[i],
+                        conv_type='Conv2d',
+                        kernel_size=patch_size,
+                        stride=patch_size,
+                        padding='corner',
+                        norm_cfg=norm_cfg,
+                        input_size=input_size,
+                        init_cfg=None))
+
+            self.pos_drops.append(nn.Dropout(p=drop_rate))
+
+        # transformer encoder
+        dpr = [
+            x.item() for x in torch.linspace(0, drop_path_rate, sum(depths))
+        ]  # stochastic depth decay rule
+        cur = 0
+
+        for k in range(len(depths)):
+            _block = ModuleList([
+                GSAEncoderLayer(
+                    embed_dims=embed_dims[k],
+                    num_heads=num_heads[k],
+                    feedforward_channels=mlp_ratios[k] * embed_dims[k],
+                    attn_drop_rate=attn_drop_rate,
+                    drop_rate=drop_rate,
+                    drop_path_rate=dpr[cur + i],
+                    num_fcs=2,
+                    qkv_bias=qkv_bias,
+                    act_cfg=dict(type='GELU'),
+                    norm_cfg=dict(type='LN'),
+                    sr_ratio=sr_ratios[k]) for i in range(depths[k])
+            ])
+            self.blocks.append(_block)
+            cur += depths[k]
+
+        self.norm_name, norm = build_norm_layer(
+            norm_cfg, embed_dims[-1], postfix=1)
+
+        self.out_indices = out_indices
+        self.extra_norm = extra_norm
+        if self.extra_norm:
+            self.norm_list = ModuleList()
+            for dim in embed_dims:
+                self.norm_list.append(build_norm_layer(norm_cfg, dim)[1])
+
+        self.pos_block = ModuleList([
```

Comment:
```suggestion
        self.position_encodings = ModuleList([
```
and move this part to the below of patch_embeds.